nix/shell.nix

```
@@ -0,0 +1,20 @@
+{ self, nixpkgs, flake-utils, ... }:
+flake-utils.lib.eachDefaultSystem (system:
+  let
+    pkgs = import nixpkgs {
+      inherit system;
+    };
```

```suggestion
    pkgs = nixpkgs.legacyPackages.${system};
```

With import, you're re-evaluating nixpkgs, so it should be avoided when possible.